cumulusci/tasks/bulkdata/load.py

```diff
@@ -508,3 +548,128 @@ def _expand_mapping(self):
+                return
+
+    def _filter_out_person_account_records(self, query, model):
+        return query.filter(model.__table__.columns.get("IsPersonAccount") == "false")
+
+    def _get_account_id_lookup(self, mapping):
+        """
+        Returns "AccountId" lookup if it exists.
+        """
+        for api_name, lookup in mapping.get("lookups", {}).items():
+            if api_name.lower() == "accountid":
+                return lookup
+
+    def _generate_contact_id_map_for_person_accounts(
+        self, contact_mapping, account_id_lookup, conn
+    ):
+        """
+        Yields (local_id, sf_id) for Contact records where IsPersonAccount
+        is true that can handle large data volumes.
+
+        We know a Person Account record is related to one and only one Contact
+        record.  Therefore, we can map local Contact IDs to Salesforce IDs
+        by previously inserted Account records:
+        - Query the DB to get the map: Salesforce Account ID ->
+          local Contact ID
+        - Query Salesforce to get the map: Salesforce Account ID ->
+          Salesforce Contact ID
+        - Merge the maps
+        """
+        # Contact table columns
+        contact_model = self.models[contact_mapping.get("table")]
+
+        contact_id_column = getattr(
+            contact_model, contact_model.__table__.primary_key.columns.keys()[0]
+        )
+        account_id_column = getattr(
+            contact_model, account_id_lookup.get_lookup_key_field(contact_model)
+        )
+
+        # Account ID table + column
+        account_sf_ids_table = account_id_lookup["aliased_table"]
+        account_sf_id_column = account_sf_ids_table.columns["sf_id"]
+
+        # Query the Contact table for person account contact records so we can
+        # create a Map: Account SF ID --> Contact ID.  Outer join the
+        # Account SF IDs table to get each Contact's associated
+        # Account SF ID.
+        query = (
+            self.session.query(contact_id_column, account_sf_id_column)
+            .filter(contact_model.__table__.columns.get("IsPersonAccount") == "true")
+            .outerjoin(
+                account_sf_ids_table,
+                account_sf_ids_table.columns["id"] == account_id_column,
+            )
+        )
+
+        # Stream the results so we can process batches of 200 Contacts
+        # in case we have large data volumes.
+        query_result = conn.execution_options(stream_results=True).execute(
+            query.statement
+        )
+
+        while True:
+            # While we have a chunk to process
+            chunk = query_result.fetchmany(200)
+            if not chunk:
+                break
+
+            # Collect Map: Account SF ID --> Contact ID
+            contact_ids_by_account_sf_id = {record[1]: record[0] for record in chunk}
+
+            # Query Map: Account SF ID --> Contact SF ID
+            # It's safe to use query_all since the chunk size to 200.
```

Comment:
The WHERE clause limit is 4,000 characters. Each Id takes 20 characters (18 + two quotes), plus N-1 commas, plus the 44 static characters = 44 + 199 + (20 * 200) = 4,243 characters. The maximum we can fetch is 188 records, if my math's right, unless we switch to using the Collections API.

Reply:
Per @spelak-salesforce this works fine with volumes of 25k contacts in batches of 200, so I'm resolving - unclear why this is not tripping a SOQL limit.

Reply:
For reference, the WHERE clause length is 4,247 using `WHERE IsPersonAccount = true AND AccountId IN ('0018A00000VFVsIQAX','0018A00000VFVsJQAX',...)` as the length of the string.  (Maybe the `WHERE` is not counted?).

Either way, integration tests pass, so it seems we don't need to worry about this limit in query filter characters.